libsolidity/parsing/Parser.cpp

```diff
@@ -332,14 +332,18 @@ StateMutability Parser::parseStateMutability(Token::Value _token)
 	return stateMutability;
 }
 
-Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(bool _forceEmptyName, bool _allowModifiers)
+Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(
+	bool _forceEmptyName,
+	bool _allowModifiers,
+	ASTString const* _contractName
```

Comment:
Sorry, didn't notice this function didn't had access to the contract name. Since this should be removed in a couple of releases, probably it doesn't matter which way it is.

@chriseth which way do you prefer, with this or without this commit?

Reply:
I don't care.

Reply:
Perhaps this one is silghtly better because of the locality of the `isConstructor` property.